libs/langgraph/langgraph/pregel/_loop.py

```diff
@@ -1048,7 +1048,35 @@ def put_writes(self, task_id: str, writes: WritesT) -> None:
 
     def __enter__(self) -> Self:
         if self.checkpointer:
-            saved = self.checkpointer.get_tuple(self.checkpoint_config)
+            if not self.is_nested:
+                saved = self.checkpointer.get_tuple(self.checkpoint_config)
+            else:
+                # when rewinding into a nested graph, ignore checkpoints newer than the
+                # boundary id so the subgraph reruns instead of replaying future state.
+                boundary_checkpoint_id = self.checkpoint_config["metadata"].get(
+                    "checkpoint_id"
+                )
+                checkpoint_ns: str = self.checkpoint_config["configurable"].get(
+                    "checkpoint_ns", ""
+                )
+
+                config = self.checkpoint_config.copy()
+                config["configurable"]["checkpoint_ns"] = checkpoint_ns
+
+                checkpoints = self.checkpointer.list(config)
```

Comment:
We will most definitely not do this, checkpoint ids are monotonically increasing, so there’s no reason why you’d ever need to list them in the execution path.